### PR TITLE
feat(site): Add source code tab on template page, group buttons and add edit file option

### LINKED ISSUE #6119: template editor: add "edit" button and code preview to template page, not tied to a specific version
I'd love to see the source code and edit a template without having to go into the latest version. 
Screenshot 2023-02-08 at 3 30 40 PM

### PR DESCRIPTION
**Pull request by BrunoQuaresma**
Closes #6119 

Demo:

https://user-images.githubusercontent.com/3165839/226437883-2babefab-7371-439f-9750-b0f5a081f5fe.mov

### CODE REVIEW

**Review by bpmct — COMMENTED** [review]
Love it! Thoughts on showing the current version name in the source code tab and a link to see all versions so a user could quickly switch?

**Comment by matifali** [comment]
Personally, The buttons looked better and more accessible to me.

**Comment by BrunoQuaresma** [comment]
@matifali yes, but it does not scale very well for now. I will try to group some of these features and see if by reducing the number of actions on the header, we can display them again.

File: site/src/components/TemplateLayout/TemplateLayout.tsx [hunk 1]
```diff
@@ -50,38 +70,35 @@ export const TemplateLayout: FC<{ children?: JSX.Element }> = ({
...
 
-  if (!template || !templatePermissions) {
+  if (templateData.isLoading) {
     return <Loader />
   }
 
+  // Make typescript happy
+  if (!templateData.data) {
+    return <></>
+  }
```
Comment (Kira-Pilot) [inline site/src/components/TemplateLayout/TemplateLayout.tsx:93]: Should we just combine this with the `loading` case up above?

Comment (BrunoQuaresma) [reply site/src/components/TemplateLayout/TemplateLayout.tsx:93]: Yeah, probably better!

**Review by Kira-Pilot — APPROVED** [review]
nice work!

**Pull request merged**